src/reanimated2/hook/useAnimatedSensor.ts

```diff
@@ -146,5 +146,5 @@ export function useAnimatedSensor(
     };
   }, [sensorType, userConfig]);
 
-  return ref.current;
+  return ref.current as AnimatedSensor<ValueRotation> | AnimatedSensor<Value3D>;
```

Comment:
Can we set the appropriate return type based on `sensorType` value?

Reply:
Would need to make this function generic in regard to the `sensorType` - I don't think this would make it actually simpler.

Reply:
`ValueRotation` and `Value3D` have different sets of fields and our users deserve better type-checking and autocomplete in their code.

Reply:
Yes, but this overload is not seen by the user - it's only the implementation. He will get the type that is provided by two previous overloads.

Reply:
You're right, I missed overloads. All good!